File: backend/LearningDashboard.Api/Entities/ActivityLog.cs
namespace LearningDashboard.Api.Entities;

public class ActivityLog
{
    public int Id { get; set; }

    public int TaskId { get; set; }

    public ProjectTask Task { get; set; } = null!;

    public ActivityAction Action { get; set; }

    public string Message { get; set; } = string.Empty;

    public int? PerformedByUserId { get; set; }

    public User? PerformedByUser { get; set; }

    public DateTime CreatedAt { get; set; }
}

File: frontend/src/types/activityLog.ts
export interface ActivityLogEntry {
  id: number
  taskId: number
  action: ActivityAction
  message: string
  performedByUserId: number | null
  performedByUserName: string | null
  createdAt: string
}

export type ActivityAction = 'Created' | 'Updated' | 'StatusChanged'
